code/script.js

```diff
@@ -218,108 +224,178 @@ const generateBoard = () => {
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
+
+//This function adds to counter when a question has been asked
+const add = () => {
+  counter += 1;
+};
+
+//This function is invoked when the restart button is pressed. The whole board is generated, with all the characters
+const playAgain = () => {
+  winOrLoseSection.style.display = "none";
+  board.style.display = "";
+};
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  charactersInPlay = CHARACTERS;
+  playAgain();
+  generateBoard();
+  setSecret();
+  counter = 0;
+  questionsAsked.innerHTML = `
+  ${counter}`;
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
+  const category = questions.options[questions.selectedIndex].parentNode.label;
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // variable that stores the actual value of the question we've selected.
+  const value = questions.options[questions.selectedIndex].value;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
+  const { category, value } = currentQuestion;
+  document.getElementById("questions").value = ""; // This clears the select attribute option between guesses
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "hair" || category === "eyes") {
+    if (secret[category] === value) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "accessories" || category === "other") {
+    if (secret[category].includes(value)) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
   }
-}
+  add();
+  questionsAsked.innerHTML = `
+  ${counter}`;
+};
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value, text } = currentQuestion;
```

Comment:
Here I'm not sure what the 'text' inside the const currentQuestion is, I wasn't able to find what it might be referring to?

Reply:
I was playing around with using value and text but decided to only use value and removed all 'text' but forgot this one ;)